includes/validation/class-amp-validation-manager.php

```diff
@@ -755,21 +755,8 @@ function( $post ) {
 	 * @return void
 	 */
 	public static function add_rest_api_fields() {
-		if ( amp_is_canonical() ) {
-			$object_types = get_post_types_by_support( 'editor' ); // @todo Shouldn't this actually only be those with 'amp' support, or if if all_templates_supported?
-		} else {
-			$object_types = array_intersect(
-				get_post_types_by_support( 'amp' ),
-				get_post_types(
-					[
-						'show_in_rest' => true,
-					]
-				)
-			);
-		}
-
 		register_rest_field(
-			$object_types,
+			AMP_Post_Type_Support::get_post_types_for_rest_api(),
```

Comment:
Is this not supposed to be just `'post'`? I don't see the post types as being allowed for this function: https://developer.wordpress.org/reference/functions/register_rest_field/

Reply:
They are actually, the field will be applied as an additional property [for each object type](https://github.com/WordPress/wordpress-develop/blob/86cbde382a0d323480a14b53c86f274599880ee2/src/wp-includes/rest-api.php#L130) supplied.